hashicorp/flight
# Export script (Node version) - Building (PR)

## :pushpin: Summary

Created a "build" script that takes the SVG files exported from Figma via the "sync" script (see #164) and generates a set of "bundles" ready to be published as NPM modules, plus updates the Ember Addon SVG sprite and catalog.json files.

I have also re-organized/updated the documentation (README/CONTRIBUTING) in the project root, and the `flight-icons` and `ember-flight-icons` to make it more in sync with the current status of the project.

_Notice: this solves #171._

## :hammer_and_wrench: Detailed Description

### How to review this PR.

Probably, looking at the diff in GitHub can be difficult, there are a lot of changes so it may be difficult to compare the before and after. My suggestion is to checkout the branch, install the dependencies and run the "build" script in your local environment (see "How to Build and Test the Change" below) to see how it works and its output.

### Things to look at:

- the overall content/organization of the folder `flight-icons`
- the main script`fligth-icons/scripts/build.ts` and its sub-modules in the folder `fligth-icons/scripts/build-parts`
- the format of the generated files in the bundles (expecially the optimized SVGs and the SVG sprite)

## :building_construction: How to Build and Test the Change

- Checkout the branch of this PR
- Install the dependencies: `cd flight-icons/scripts && yarn install`
- Create a `.env` file in the `scripts` folder with your Figma API key (see CONTRIBUTING.md in the /flight-icons/ folder)
- Run the "sync" script (optional): `yarn sync`
- Run the " build" script: `yarn build`
- Check the generated files in the `dist` folder

## :+1: Definition of Done

See #171

## 🤔 Things to decide

- Do we want to commit the `dist` folder in the repo? In theory is not needed, it can always be regenerated from the `src` in any moment.
- The final format of the "custom" Consul CSS file is still under discussion, but this does not change the overall script (it's a matter of convention/agreement with the Consul team on how they want the SCSS generated).

### :speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

Examples: 
- issue (ux,non-blocking): These buttons should be red, but let's handle this in a follow-up.
- suggestion (non-blocking): Let's change this wording to make it easier to understand.
- issue (blocking): We shouldn't introduce this kind of tech debt; let's pair and resolve the issue in a more sustainable way.
